optuna/samplers/nsgaii/_crossovers/_vsbx.py

```diff
@@ -52,37 +90,47 @@ def crossover(
             eta = self._eta
 
         us = rng.rand(len(search_space_bounds))
-        beta_1 = np.power(1 / 2 * us, 1 / (eta + 1))
-        beta_2 = np.power(1 / 2 * (1 - us), 1 / (eta + 1))
-        mask = us > 0.5
-        c1 = 0.5 * ((1 + beta_1) * parents_params[0] + (1 - beta_1) * parents_params[1])
-        c1[mask] = (
-            0.5 * ((1 - beta_1) * parents_params[0] + (1 + beta_1) * parents_params[1])[mask]
-        )
-        c2 = 0.5 * ((3 - beta_2) * parents_params[0] - (1 - beta_2) * parents_params[1])
-        c2[mask] = (
-            0.5 * (-(1 - beta_2) * parents_params[0] + (3 - beta_2) * parents_params[1])[mask]
-        )
-
-        # vSBX applies crossover with establishment 0.5, and with probability 0.5,
+        beta_1 = np.power(1 / (2 * us), 1 / (eta + 1))
```

Comment:
Although it is a very small probability, there is a chance of becoming the denominator in `1 / (2 * us)` can be zero since `rng.rand()` $\in [0, 1)$ ([ref](https://numpy.org/doc/stable/reference/random/generated/numpy.random.rand.html)).

The below might be safer. (This pseudocode might require modification since us is vector whereas eps is scalar.)
```
        beta_1 = np.power(1 / max((2 * us), eps), 1 / (eta + 1))
```

Reply:
There is no strong reason for the `eps`, but I have chosen to use the same values used in the SBX.

https://github.com/optuna/optuna/blob/19296c387c5376c137f9c3afc4d18159e70bdc9c/optuna/samplers/nsgaii/_crossovers/_sbx.py#L99

Although out of the range of values actually generated, 1 is set as us value in the test, and since zero divisions occur even in beta_2, a similar correction has been made.

```
eta_2 = np.power(1 / (2 * (1 - us)), 1 / (eta + 1))
```